packages/lib/response/service.ts

```diff
@@ -246,75 +235,131 @@ export const getResponse = async (responseId: string): Promise<TResponse | null>
   }
 };
 
-export const getSurveyResponses = async (surveyId: string, page?: number): Promise<TResponse[]> => {
-  validateInputs([surveyId, ZId], [page, ZOptionalNumber]);
+export const getResponse = async (responseId: string): Promise<TResponse | null> =>
+  unstable_cache(
+    async () => {
+      validateInputs([responseId, ZId]);
 
-  try {
-    const responses = await prisma.response.findMany({
-      where: {
-        surveyId,
-      },
-      select: responseSelection,
-      orderBy: [
-        {
-          createdAt: "desc",
-        },
+      try {
+        const responsePrisma = await prisma.response.findUnique({
+          where: {
+            id: responseId,
+          },
+          select: responseSelection,
+        });
+
+        if (!responsePrisma) {
+          throw new ResourceNotFoundError("Response", responseId);
+        }
+
+        const response: TResponse = {
+          ...responsePrisma,
+          person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
+          tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+        };
+
+        return response;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getResponse-${responseId}`],
+    { tags: [responseCacheTag.single(responseId)], revalidate: SERVICES_REVALIDATION_INTERVAL }
+  )();
+
+export const getSurveyResponses = async (surveyId: string, page?: number): Promise<TResponse[]> =>
+  unstable_cache(
+    async () => {
+      validateInputs([surveyId, ZId], [page, ZOptionalNumber]);
+
+      try {
+        const responses = await prisma.response.findMany({
+          where: {
+            surveyId,
+          },
+          select: responseSelection,
+          orderBy: [
+            {
+              createdAt: "desc",
+            },
+          ],
+          take: page ? ITEMS_PER_PAGE : undefined,
+          skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
+        });
+
+        const transformedResponses: TResponse[] = responses.map((responsePrisma) => ({
+          ...responsePrisma,
+          person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
+          tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+        }));
+
+        return transformedResponses;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getSurveyResponses-${surveyId}`],
+    {
+      tags: [
+        getSurveyCacheTag(surveyId),
+        getDisplaysCacheTag(surveyId),
```

Comment:
Also displays are not a part of responses. But notes, tags and people are. But with people I'm not sure how we revalidate. I don't like to revalidate every time a new person is created or updates as this happens a lot.